ecommerce/views/v0/__init__.py

```diff
@@ -136,3 +170,441 @@ def create(self, request, *args, **kwargs):
+            self.logger.debug(
+                "Error happened submitting the transaction: {msg}".format(
+                    msg=fmt_response.message
+                )
+            )
+            template = "test_checkout_error.html"
+        elif fmt_response.state == ProcessorResponse.STATE_CANCELLED:
+            # Transaction cancelled
+            # Transaction could be cancelled for reasons that don't necessarily
+            # mean that the entire order is invalid, so we'll do nothing with
+            # the order here.
+            self.logger.debug(
+                "Transaction cancelled: {msg}".format(msg=fmt_response.message)
+            )
+
+            template = "test_checkout_cancelled.html"
+        elif fmt_response.state == ProcessorResponse.STATE_REVIEW:
+            # Transaction needs a professional review
+            # (nonjokey - someone needs to approve this so the order shouldn't be cancelled quite yet)
+            self.logger.debug(
+                "Transaction flagged for review: {msg}".format(msg=fmt_response.message)
+            )
+
+            template = "test_chekcout_review.html"
+        elif fmt_response.state == ProcessorResponse.STATE_ACCEPTED:
+            # It actually worked here
+            self.logger.debug(
+                "Transaction accepted!: {msg}".format(msg=fmt_response.message)
+            )
+
+            order.state = Order.STATE.FULFILLED
+            order.save()
+
+            template = "test_checkout_accept.html"
+
+        return render(
+            request,
+            template,
+            {
+                "order": order,
+                "fmt_response": fmt_response,
+                "full_response": request.POST,
+            },
+        )
+
+
+"""
+These are testing stubs - they will allow you to fairly easily run a transaction
+through the whole workflow. These should be replaced with the actual workflow
+when that's more ready to go. (They should be removed when there's real UI.)
+"""
+
+
+class CheckoutTestView(TemplateView):
+    template_name = "test_checkout_post.html"
+
+    def get(self, request):
+        """
+        Generates a basket, and then adds a program to it, then gives you a form
+        so you can check out through the payment processor.
+        """
+        import random
+
+        if not request.user.is_superuser:
+            raise PermissionDenied("Not an admin user.")
+
+        if Product.objects.count() == 0:
+            raise Exception("You need to set up a product first.")
+
+        if Basket.objects.filter(user=request.user).count() == 0:
+            basket = Basket(user=request.user)
+            basket.save()
+        else:
+            basket = Basket.objects.filter(user=request.user).get()
+
+        if basket.basket_items.count() == 0:
+            products = Product.objects.all()
+
+            basket_item = BasketItem(
+                product=products[random.randrange(1, len(products), 1)],
```

Comment:
Knit, this errors if there's only 1 product.
```suggestion
                product=products[random.randrange(0, len(products), 1)],
```